source/faq.txt

```diff
@@ -19,7 +19,87 @@ This page contains frequently asked questions and their corresponding answers.
    If you can't find an answer to your problem on this page,
    see the :ref:`golang-issues-and-help` page for next steps and more
    resources.
+
+.. _golang-faq-connection-pool:
+
+How Does Connection Pooling Work in the {+driver-short+}?
+---------------------------------------------------------
+
+Every ``Client`` instance has a built-in connection pool for each server
+in your MongoDB topology. These pools open sockets on demand to support
+concurrent MongoDB operations, or `goroutines
+<https://www.golang-book.com/books/intro/10>`__, in your application.
+
+The maximum size of each connection pool is set by ``maxPoolSize``, which
+defaults to ``100``. If the number of connections to a server is
+the value of ``maxPoolSize`` and every connection is in use, the next
+request to that server will wait until one of the connections becomes
+available.
+
+The ``Client`` instance opens two additional sockets per server in your
+MongoDB topology for monitoring the server's state.
+
+For example, a client connected to a 3-node replica set opens 6
+monitoring sockets. It also opens as many sockets as needed to support
+an application's concurrent operations on each server, up to
+the value of ``maxPoolSize``. If ``maxPoolSize`` is ``100`` and the
+application only uses the primary (the default), then only the primary
+connection pool grows and there can be at most ``106`` total connections. If the
+application uses a :ref:`read preference <golang-read-pref>` to query the
+secondary nodes, their pools also grow and there can be ``306`` total connections.
+
+Additionally, connection pools are rate limited such that each connection pool
+can only create, at maximum, the value of ``maxConnecting`` connections
+in parallel at any time. Any additional goroutine stops waiting in the
+following cases:
+
+- One of the existing goroutines finishes creating a connection, or
+  an existing connection is checked back into the pool.
+- Rate limiting concurrent connection creation reduces the likelihood of
+  connection storms and improves the driver's ability to reuse existing
+  connections.
```

Comment:
I'm a little confused at this bullet. I was expecting to see a situation in which a goroutine will stop waiting for a connection, but this seems more like a separate fact about rate limiting. I'm not entirely sure how it relates to the goroutine waiting. 

Could this be reworded to make the focus of the bullet be the action/situation that prompts the goroutine to stop waiting?

Reply:
How is this attempt?